### ISSUE TITLE
[sig-node] Failure cluster [c13abd3c...] flakes from Memory Manager Metrics

### ISSUE THREAD

**Issue by dims** [kubernetes/kubernetes#123454]
##### Error text:
```
[FAILED] Timed out after 60.000s.
Expected
    <string>: KubeletMetrics
to match keys: {
."kubelet_memory_manager_pinning_requests_total"[]:
	Expected
	    <string>: Sample
	to match fields: {
	.Value:
		Expected
		    <model.SampleValue>: 2
		to be ==
		    <int>: 1
	}
	
}
In [It] at: k8s.io/kubernetes/test/e2e_node/memory_manager_metrics_test.go:122 @ 02/21/24 06:40:31.854

```
#### Recent failures:
[2/22/2024, 7:30:31 AM ci-cgroupv2-containerd-node-arm64-e2e-serial-ec2](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-cgroupv2-containerd-node-arm64-e2e-serial-ec2/1760642962640867328)
[2/22/2024, 4:10:17 AM ci-kubernetes-node-kubelet-serial-containerd](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-serial-containerd/1760592630795735040)
[2/22/2024, 1:33:30 AM ci-cgroupv2-containerd-node-arm64-al2023-e2e-serial-ec2-eks](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-cgroupv2-containerd-node-arm64-al2023-e2e-serial-ec2-eks/1760552867929788416)
[2/22/2024, 12:09:14 AM ci-kubernetes-node-kubelet-serial-containerd](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-serial-containerd/1760531981122146304)
[2/21/2024, 10:00:20 PM ci-kubernetes-node-arm64-ubuntu-serial](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-node-arm64-ubuntu-serial/1760499516097695744)

/kind failing-test

/sig node

**Comment by dims** [comment]
tests were added in https://github.com/kubernetes/kubernetes/pull/121778

/assign @Tal-or 
/cc @swatisehgal @ffromani

**Comment by Tal-or** [comment]
@dims thank you for raising that up.

I'll be back in the middle of next week and get to it ASAP

**Comment by pacoxu** [comment]
https://testgrid.k8s.io/sig-node-release-blocking#node-kubelet-serial-containerd

This flakes in sig node release blocking testgrid. 

I see @ffromani 's PR #123468 is fixing it. Can that fix target v1.30?

**Comment by ffromani** [comment]
> https://testgrid.k8s.io/sig-node-release-blocking#node-kubelet-serial-containerd
> 
>     * `[sig-node] Memory Manager Metrics [Serial] [Feature:MemoryManager] when querying /metrics should report zero pinning counters after a fresh restart`
> 
> 
> This flakes in sig node release blocking testgrid.
> 
> I see @ffromani 's PR #123468 is fixing it. Can that fix target v1.30?

I guess we can. I'm monitoring the lanes and I'm willing to keep improving the test reliability.
